src/core/grid_based_algorithms/lb_interface.cpp

```diff
@@ -563,28 +564,36 @@ void lb_lbfluid_print_vtk_velocity(const std::string &filename,
     throw std::runtime_error("Could not open file for writing.");
   }
 
-  std::vector<int> bb_low;
-  std::vector<int> bb_high;
+  Utils::Vector3i bb_low{0, 0, 0};
+  Utils::Vector3i bb_high{0, 0, 0};
+  if (lattice_switch == ActiveLB::GPU) {
+#ifdef CUDA
+    bb_high = {static_cast<int>(lbpar_gpu.dim_x) - 1,
+               static_cast<int>(lbpar_gpu.dim_y) - 1,
+               static_cast<int>(lbpar_gpu.dim_z) - 1};
+#endif //  CUDA
+  } else {
+    bb_high = {lblattice.global_grid[0] - 1, lblattice.global_grid[1] - 1,
+               lblattice.global_grid[2] - 1};
+  }
```

Comment:
```suggestion
  Utils::Vector3i bb_low{0, 0, 0};
  auto const bb_high = lb_lbfluid_get_shape() - Utils::Vector3i{1,1,1};
```

Reply:
here, we can probably drop the - 1 and just check for smaller

Reply:
good idea!

Reply:
done